src/server.ts

```diff
@@ -124,36 +123,40 @@ function browsersListParser(data: string): string[] {
 		.map((line: string) => line.trim());
 }
 
-function getBrowsersList(documentFsPath: string): Promise<string[]> {
+function getBrowsersList(document: string): Promise<string[]> {
 	const configResolverOptions: IOptions = {
 		packageProp: 'browserslist',
 		configFiles: [
             '.browserslistrc',
 			'browserslist'
 		],
 		editorSettings: workspaceSettings.browsers || null,
-		parsers: [
-			{ pattern: /\.?browserslist(rc)?$/, parser: browsersListParser }
-		]
+		parsers: [{
+			pattern: /\.?browserslist(rc)?$/,
+			parser: browsersListParser
+		}]
 	};
 
-	if (!needUpdateConfig) {
-		return Promise.resolve(browsersListCache);
+	if (browsersListCache[document]) {
+		return Promise.resolve(browsersListCache[document]);
 	}
 
 	return configResolver
```

Comment:
JFYI: You can use the [config-profiler](https://github.com/mrmlnc/config-profiler) package that has a cache out of the box. Just for for the future improvements.

Reply:
This is not really a cache, rather a store for all the different `browserslist` configs there are on the different folder paths. I believe most user only uses a single setting, since the general use case is to develop to one scope. However, it can happen that for some reason different parts of the code base support different scopes.

Reply:
- [ ] Store `browserslist` by current path rather than by document. This should better describe what is happening as well as, it would reduce duplicates in the store.
- [ ] Rename cache to store